src/ln.js

```diff
@@ -44,8 +42,14 @@ function _ln(options, source, dest) {
   }
 
   if (options.symlink) {
+    if ((isAbsolute && !fs.existsSync(sourcePath)) || !fs.existsSync(path.resolve(process.cwd(), path.dirname(dest), source))) {
+      common.error('Source file does not exist', true);
+    }
     fs.symlinkSync(source, dest, os.platform() === "win32" ? "junction" : null);
   } else {
+    if (!fs.existsSync(source)) {
+      common.error('Source file does not exist', true);
+    }
```

Comment:
In testing, I found that the source file was not checked properly for hard links.This correctly checks if the source exists for hard links.